engine/runner/src/main/scala/org/enso/runner/Main.scala

```diff
@@ -843,11 +843,37 @@ object Main {
     */
   val defaultLogLevel: LogLevel = LogLevel.Error
 
+  var sampler : org.netbeans.modules.sampler.Sampler = org.netbeans.modules.sampler.Sampler.createSampler("LanguageServer")
+
+  def stopSampling(): Unit = {
+    if (sampler == null) {
+        return;
+    }
+    val where = "/tmp/language-server.npss"
+    System.err.println("Dumping out the sampled data " + where)
+    val out = new java.io.FileOutputStream(where)
+    val data = new java.io.DataOutputStream(out)
+    sampler.stopAndWriteTo(data)
+    data.close()
+    System.err.println("Dumping done: " + where)
+    sampler = null;
+  }
+
   /** Main entry point for the CLI program.
     *
     * @param args the command line arguments
     */
   def main(args: Array[String]): Unit = {
+    sampler.start();
+    System.err.println("Sampling has been started: " + sampler)
+
+    new java.util.Timer().schedule(new java.util.TimerTask() {
+        def run(): Unit = {
+            System.err.println("Timeout 10s!")
+            stopSampling()
+        }
+    }, 10000);
```

Comment:
minor issue - the error hardcodes 10s, the same value as provided later - can we refactor it to a variable and use in both places, please?

Reply:
@JaroslavTulach you can add a flag `--profiling=<time interval>` that enables profiling for requested period of time